### ISSUE TITLE
Flickering with onLoop gotoFrame

### ISSUE THREAD

**Issue by jaythomas** [kikito/anim8#19]
When an animation is done with a full play, I want to rewind back to a mid-point in the animation and loop that indefinitely. My intent behind this is to avoid creating separate transition sprite animations. In order to do this, I am setting the gotoFrame in the onLoop function. Given I have 4 frames and want to loop 3-4 after the initial loop, my code looks like this:

```lua
  local onLoop = function(animation)
    animation:gotoFrame(3)
end
```

The problem is the animation very briefly (for one draw iteration) is set back to frame 2. You can visually see this issue in my example [here](https://github.com/jaythomas/anim8/tree/issue-19). The set duration doesn't affect the results.

If I'm not using this appropriately then apologies.

**Comment by kikito** [comment]
Hi, thanks for reporting this and taking the time to write a demo of the problem.

It does seem like there is a bug somewhere. I will investigate.

**Comment by jaythomas** [comment]
It looks like the problem originates [here](https://github.com/kikito/anim8/blob/539cc26fee6517a37448ee70ee830e56921137be/anim8.lua#L237). The position is being set inside the update *after* the onLoop callback is invoked and attempting to set the position. Moving it above like so will workaround the issue and eliminate the flicker:
```diff
diff --git a/anim8.lua b/anim8.lua
index 8fefe99..e8763f6 100644
--- a/anim8.lua
+++ b/anim8.lua
@@ -227,14 +227,14 @@ function Animation:update(dt)
   if self.status ~= "playing" then return end

   self.timer = self.timer + dt
+  self.position = seekFrameIndex(self.intervals, self.timer)
+
   local loops = math.floor(self.timer / self.totalDuration)
   if loops ~= 0 then
     self.timer = self.timer - self.totalDuration * loops
     local f = type(self.onLoop) == 'function' and self.onLoop or self[self.onLoop]
     f(self, loops)
   end
-
-  self.position = seekFrameIndex(self.intervals, self.timer)
 end

 function Animation:pause()
```

but there is possibly an ill side-effect to doing this that I haven't discovered yet. Applying this patch in my game and doing a quick smoke test I don't see any problems in changing the ordering. Let me know what you think.

**Comment by tapir** [comment]
Have you noticed any side effects?

**Comment by jaythomas** [comment]
@tapir, the position needs to update but at the moment I can't remember why. If you look at the pull request in the link you will see I have a pull request that implements a different solution so that position can be called after the onLoop, but only when needed.

**Comment by kikito** [comment]
Hello @jaythomas , can you try the current implementation? I just merged #28 , which might solve your issue.

**Comment by kikito** [comment]
Closing this since I think it was fixed in #28. Please reopen if needed. Sorry it took so long to get a fix.

**Comment by jaythomas** [comment]
I can confirm that #28 resolves this issue, thanks.
